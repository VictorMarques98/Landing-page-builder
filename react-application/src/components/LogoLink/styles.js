/* eslint-disable no-unused-vars */
import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    text-decoration: none;
    align-items: center;
    color: inherit;

    > img {
      max-height: 5rem;
    }
  `}
`;
