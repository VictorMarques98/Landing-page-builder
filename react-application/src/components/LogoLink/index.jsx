/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './styles';
import Heading from '../Heading';

export const LogoLink = ({ text, image, link }) => {
  return (
    <Heading size="small" uppercase>
      <Styled.Container href={link}>
        {!!image && <img src={image} alt={text} />}
        {!image && text}
      </Styled.Container>
    </Heading>
  );
};

export default LogoLink;

LogoLink.defaultProps = {
  text: '',
  link: '',
  image: '',
};

LogoLink.propTypes = {
  text: PropTypes.string.isRequired,
  image: PropTypes.string,
  link: PropTypes.string,
};
