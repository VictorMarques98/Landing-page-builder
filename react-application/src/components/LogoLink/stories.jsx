import React from 'react';
import LogoLink from '.';

export default {
  title: 'LogoLink',
  component: LogoLink,
  args: {
    text: 'LogoLink',
    image: 'assets/images/logo.png',
    link: 'http://localhost',
  },
};

export const ImageOnly = (args) => {
  return (
    <div>
      <LogoLink {...args} />
    </div>
  );
};

export const TextOnly = (args) => {
  return (
    <div>
      <LogoLink {...args} />
    </div>
  );
};

TextOnly.args = {
  image: '',
};
