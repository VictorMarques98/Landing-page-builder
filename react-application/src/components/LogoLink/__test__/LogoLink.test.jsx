import { screen } from '@testing-library/react';
import React from 'react';
import LogoLink from '../';
import renderTheme from '../../../styles/render-theme';

describe('<LogoLink />', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink text="logoLink" link="https://www.google.com.br" />);
    const logoLink = screen.getByText('logoLink');
    expect(logoLink).toBeInTheDocument();
  });

  it('should render image logo', () => {
    renderTheme(<LogoLink text="logoLink" link="#target" image="image.jpg" />);
    expect(screen.getByAltText('logoLink')).toHaveAttribute('src', 'image.jpg');
  });

  it('should match SnapShot', () => {
    const { container } = renderTheme(<LogoLink text="logoLink" link="#target" image="image.jpg" />);
    expect(container).toMatchSnapshot();
  });
});
