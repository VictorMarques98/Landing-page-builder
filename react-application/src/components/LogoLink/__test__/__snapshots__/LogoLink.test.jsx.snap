// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match SnapShot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: inherit;
}

.c1 > img {
  max-height: 5rem;
}

.c0 {
  color: #0a1128;
  text-transform: uppercase;
  font-size: 2.4rem;
  text-transform: uppercase;
}

<div>
  <h1
    class="c0"
  >
    <a
      class="c1"
      href="#target"
    >
      <img
        alt="logoLink"
        src="image.jpg"
      />
    </a>
  </h1>
</div>
`;
